feature_engine/variable_manipulation.py

```diff
@@ -110,15 +110,23 @@ def _find_or_check_categorical_variables(
         variables = [variables]
 
     elif not variables:
-        variables = list(X.select_dtypes(include=["O", "category"]).columns)
+        variables = list(
```

Comment:
The reason is: that function detected datetime variables that were disguised as object as categorical variables. E.g. the column "data" in pd.DataFrame({"data" : ['10/03/21', '05/02/09']}) <br>
Now I believe that was fine up until we didn't have any datetime transformers, but now that datetime transformers are looking to convert such variables into datetime to perform downstream tasks such as feature extraction, I thought categorical encoders had better *not* work on those features, for the sake of consistency.

Otherwise, if say a categorical encoder was __init__'d with variables=None and a datetime transformer was also __init__'d with variables=None, both would work on those variables whose dtype is object but actually contain datetime information.

Hope I made myself clearer now, obviously it's a design choice and it's completely up to you to decide what to do with it

Reply:
Thanks for explaining. I think let's not mix functions and let's keep checking datetime to a different function.

Reply:
Okay, although I believe this will result in requiring the user to be more diligent when initializing the transformer (i.e., passing reasonable variables), otherwise stuff like this could happen:

1. user calls a categorical encoder on a "datetime" variable (e.g. by passing variables=None, and _find_or_check_categorical_variables will consider it a categorical variable if its dtype is object), and the variable gets encoded and its dtype changed to some numerical type in the process.
2. user then calls a datetime transformer on the same variable (either by passing variables=None or specifying it), but since the previous call to the categorical transformer encoded that variable into a numerical feature, the datetime transformer will not be able to process it, thus the actual datetime information on that variable is lost

This sounds potentially dangerous to me, but if you're okay with it then I will revert any changes made on that function

Reply:
I see your point. Let me sleep on it

Reply:
I see your point here, and I think it makes sense to ignore datetime variables which type is object. But I think we should do that in a different PR. 

I created a new issue #336